### DISCUSSION TITLE
Lesson 7 How to save a bit more gas

### DISCUSSION THREAD

**Discussion by pyromaus** [Cyfrin/foundry-full-course-cu#175] [Ideas]
A couple ways to save a tiny bit more gas in Lesson 7 withdraw functions.

0. Initial numbers; 780 gas saved by using `uint fundersLength = s_funders.length` in `cheaperWithdraw()` (as per the lesson)
(actually 764 gas - `testWithdrawMultipleFundersCheaper()` seems to take 16 more gas with identical code. I guess from the word "cheaper" present in both the withdraw function and the test function)
```solidity
[PASS] testWithdrawMultipleFunders() (gas: 490167)
[PASS] testWithdrawMultipleFundersCheaper() (gas: 489387)
```

1. add `address[] memory funderz = s_funders;` to `cheaperWithdraw()` and refer to it in the `for` loop: 
`address funder = funderz[funderIndex]`. 
```solidity
[PASS] testWithdrawMultipleFunders() (gas: 490167)
[PASS] testWithdrawMultipleFundersCheaper() (gas: 489187)
```
+200 gas savings

2. change `vm.startPrank()` to `vm.prank()` and remove `vm.stopPrank()` in `testWithdrawMultipleFundersCheaper()`
(gas spent on tests doesn't rly matter but I found it interesting)
```solidity
[PASS] testWithdrawMultipleFunders() (gas: 490167)
[PASS] testWithdrawMultipleFundersCheaper() (gas: 488906)
```
+281 gas savings. 481 total

**Comment by PatrickAlphaC** [comment]
LOVE this 🔥🔥🔥🔥🔥

**Comment by conjohnerton** [comment]
That second one is really good! Also popped into my head during the lesson. If there were a way to reinit storage mappings, there might be even more gas optimization there, alas I couldn't find a way to do that in a single operation.

**Comment by theirrationalone** [comment]
Seems like an excerpt from Patrick's blockchain HH.
